lib/fs.js

```diff
@@ -15,6 +15,10 @@ module.exports = {
 
     for (const compiler of compilers) {
       compiler.hooks.emit.tap('WebpackDevMiddleware', (compilation) => {
+        if (compiler.hasWebpackDevMiddlewareAssetEmittedCallback) {
```

Comment:
Ugly name, but we guarantee what we don't intersect with something

Reply:
Also in webpack@5 `assetEmitted` hook has `compilation` in arguments so we simplify this code in near future

Reply:
`compiler.hasWebpackDevMiddlewareAssetEmitted` is enough to make the point.